### ISSUE TITLE
New feature: validate --shacl

### ISSUE THREAD

**Issue by matentzn** [ontodev/robot#576]
I saw some cool uses of shacl as a constraint language for RDF graphs at the tutorial at Connected Data 2019; I think it would be v. cool if we could offer to do this with ROBOT.

Here is the library: https://github.com/TopQuadrant/shacl

The license seem to be apache 2, not sure what that means for us. Maybe @HolgerKnublauch could tell us whether this is at all allowed, and we can think whether it makes sense to support it given our relatively large user base. One possible way to phrase this command would be

```
robot validate -i ont.owl --type shacl --schema shapes.shacl 
```
or rely on file extension to disambiguate.

If this is overkill for ROBOT, it could also go straight into ODK (license permitted). @cmungall @jamesaoverton @balhoff @beckyjackson

**Comment by balhoff** [comment]
I agree, this would be a handy extension to `validate`. I would also like to include ShEx, which is what we have been using recently in GO (https://github.com/geneontology/go-shapes). There is a Java library at (https://github.com/iovka/shex-java). @cmungall started on a general OBO validation ShEx schema, and I have some improvements to it that I need to commit. I wonder if this is getting too kitchen sink? Should there just be independent SHACL/ShEx tools? I lean toward including them though.

**Comment by HolgerKnublauch** [comment]
My understanding is that Apache license basically allows whatever you like.

**Comment by jamesaoverton** [comment]
See also https://github.com/ontodev/robot/issues/387 where we're talking about a `validate` command with DL Query, mainly for tables.

I like the idea of adding SHACL and ShEx to ROBOT, as long as the dependencies are all on the JVM and not overwhelmingly large. ROBOT is already a kitchen sink. The value-added is a single JAR to install, OBO-focus, and the most coherent interface we can manage.

My only concern is that we're talking about a lot of different approaches to validation. Maybe these could all be called `validate` with command-line switches (at the risk of overloading), or maybe they get their own specialized commands, like our existing `validate-profile` (more verbose, maybe more confusing).

**Comment by matentzn** [comment]
I am not concerned about naming here; you can go ahead as you see fit. Specific commands are fine by me..

**Comment by cmungall** [comment]
The lib we would use would be java.

I think having a declarative schema for the shapes we expect for OBO
classes would be fantastic. It's great that robot report will check for
some kinds of cardinality violations but having things together in a shape
file would be great.

There will be challenges in defining what the schema is. OBO has been quite
anarchic here, which has caused problems for developers who expect a more
predictable structure. We even failed to agree on a single vocabulary for
synonyms in OBO.

I expect we will have a modular layered approach, with a loose core, and
additional modules of constraints on top. This will give us more of a
vocabulary for describing a more predictable shape of an ontology, e.g.
this ontology is obo-basic following go-like-properties-variant1, or
obi-like-properties-variant2
